File: network_frame_cpp/src/application/application.cpp
#include "application/application.h"

namespace application {

Data::Data(const std::string& payload) 
    : payload_(payload.begin(), payload.end()) {
}

Data::Data(const std::vector<uint8_t>& payload) 
    : payload_(payload) {
}

std::vector<uint8_t> Data::getPayload() const {
    return payload_;
}

std::string Data::getPayloadString() const {
    return std::string(payload_.begin(), payload_.end());
}

size_t Data::size() const {
    return payload_.size();
}

} // namespace application

File: network_frame/pkg/application/application.go
package application

// Data 表示应用层数据
type Data struct {
	Payload []byte
}

// NewData 创建新的应用层数据
func NewData(payload string) *Data {
	return &Data{
		Payload: []byte(payload),
	}
}

// NewDataFromBytes 从字节切片创建应用层数据
func NewDataFromBytes(payload []byte) *Data {
	return &Data{
		Payload: payload,
	}
}

// GetPayload 获取原始数据
func (d *Data) GetPayload() []byte {
	return d.Payload
}

// GetPayloadString 获取原始数据字符串形式
func (d *Data) GetPayloadString() string {
	return string(d.Payload)
}

// Size 获取数据大小
func (d *Data) Size() int {
	return len(d.Payload)
}
